src/librustc/mir/interpret/allocation.rs

```diff
@@ -346,11 +346,17 @@ impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
         &mut self,
         cx: &impl HasDataLayout,
         ptr: Pointer<Tag>,
-        src: &[u8],
+        src: impl IntoIterator<Item=u8, IntoIter: iter::ExactSizeIterator>,
     ) -> InterpResult<'tcx>
```

Comment:
Maybe there should be a comment suggesting to use `Memory::write_bytes` instead? I think that unless someone is writing in several calls, they should use the other function

Reply:
Well that applies to every single function in this module...

Reply:
Ok that's true